### ISSUE TITLE
Conflict between `@JsonTypeInfo.property` and property from class

### ISSUE THREAD

**Issue by vojtechhabarta** [FasterXML/jackson-databind#1633]
Jackson produces duplicate JSON property when same property name is used in `@JsonTypeInfo` annotation and data class itself.

For example this code:
``` java
public static void main(String[] args) throws Exception {
    System.out.println(new ObjectMapper().writeValueAsString(new SubType("custom")));
}

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
public class Base {}

@JsonTypeName("SubType")
public class SubType extends Base {
    public String type;
    public SubType(String type) {
        this.type = type;
    }
}
```
produces following JSON output:
``` json
{"type":"SubType","type":"custom"}
```

I tried it with Jackson versions 2.8.6, 2.8.8 and 2.9.0.pr3 (latest).

**Expected:** `type` property should be written only once but I am not sure which one should be used in this case. (Throwing exception would be too strict in my opinion.)

**Comment by cowtowncoder** [comment]
@vojtechhabarta Quick question: did you look through open issues for this one? I think we may already have a failed issue for this problem.
And yes, this is a problem, thank you for reporting it.

**Comment by vojtechhabarta** [comment]
@cowtowncoder yes, I found few issues reporting duplicate property but not in this particular case. However it is possible that I missed it.
Thanks for quick response.

**Comment by cowtowncoder** [comment]
@vojtechhabarta Ok. As long as you had a quick look that's fine -- if I find another issue I can link.
I must have been thinking of #1410 which is not the same, after all.

**Comment by cowtowncoder** [comment]
Hmmh. I should pay more attention to actual description.

So, the problem is that inclusion is defined as `As.PROPERTY`.
It needs to be defined as `As.EXISTING_PROPERTY`, to avoid duplication during serialization.
Ideally it'd be good for Jackson to try to check for existence and "just work", but the way things were designed this can not be done at least for Jackson 2.x.
